packages/taquito/src/contract/prepare.ts

```diff
@@ -339,3 +342,35 @@ export const createSmartRollupOriginateOperation = async ({
     parameters_ty: parametersType,
   } as RPCSmartRollupOriginateOperation;
 };
+
+export const createStakingOperation = async (
+  {
+    sourceAndDestination,
+    amount,
+    fee = DEFAULT_FEE.TRANSFER,
+    gasLimit = DEFAULT_GAS_LIMIT.TRANSFER,
+    storageLimit = DEFAULT_STORAGE_LIMIT.TRANSFER,
+    mutez = false,
+  }: StakingParams,
+  entrypoint: StakingEntrypoint
+) => {
+  if (!sourceAndDestination) {
+    throw new ParameterValidationError('The parameter sourceAndDestination is missing');
+  }
+  const operation: RPCTransferOperation = {
+    source: sourceAndDestination,
```

Comment:
createStakingOperation doesn't need source just like all other create...Operation. source will be added in constructOpContents

Reply:
Let's discuss this in a huddle

Reply:
@dsawali ?